Corpses left behind when killing legion have semi-randomized equipment.

Expletive opened the pull request:
:cl: Nanotrasen Mining Alert
add: Nanotrasen's mining operations have created far more corpses than an entire galaxy of cooks could hope to deal with. To solve this, we decided to just dump them all from orbit onto the barren lava planet. Unfortunately, the creatures called "Legion" have infested a great number of them, and you can now commonly find the bodies of former Nanotrasen employees left behind. Additionally, there are reports of natives, other settlers, and even stranger things found among the corpses.
/:cl:

I've tried to make any useful equipment only created rarely. Around 1-2% is what I aimed for. The vast majority of corpses will be shaft miners with nothing useful, ash walkers with only their default armor and helmet, and naked golems. Since the wish granter dungeon was removed, the syndicate hardsuit was no longer obtainable on lavaland, but with this PR it is. Similarly, with the wish granter gone, there was also no way for a miner to become a shadow without science's help.

Shaft miner corpses can have  marker beacons, and rarely have explorer's webbing, survival medipens, space cash, goliath cloaks, or very rarely a kinetic accelerator modkit.

Ashwalker corpses can semi-commonly have bone daggers, and much more rarely have tribal equipment like bone armor, sinew belts, or bone spears, or axes.

Golem corpses might have mesons or a collection of huds or welding gogles. They can also rarely have a full toolbelt or an explorer's webbing + capsule. They'll commonly have a cape of the liberator, and they'll rarely have a random advanced tool. Either a power tool or an experimental welder.

Clown corpses have the clown costume, with a high chance of a normal clown item. They can also rarely have up to 3 sheets of bananium, a golden bike horn, or sad trombone implanter in their pockets. This is blatant clowncreep and you should not stand for it.

Then the four 'special' corpses: A dead operative, primarily for the hardsuit. A dead medieval knight, with armor, a  holy claymore, and buckler,a dead shadow scientist with a pill containing 1u of shadow mutation toxin,  and finally, a cultist in old cult robes who is carrying a lot of items that would only be useful to cultists of nar'sie.

Lastly, none of these corpses will be left behind by legion that get spawned by necropolis tendrils.  Those will instead leave behind "ashen skeletons", which are naked dead skeletons with their color set to #454545.

Expletive committed:
Legion corpses have equipment. Rarely their equipment is useful.

Expletive committed:
some chance

Expletive committed:
some fixes

Steelpoint commented:
Time to camp a Legion tendril.

Limski commented:
Just casually adding yourself to the admins.txt no one will notice

Steelpoint commented:
Every single coder does that @Limski 

It does not add you to the actual server admin list, it just adds you to a coder list. So every time you download some new code you don't have to re-add yourself to the admin list.

Limski commented:
I'm not a coder and that was my first time looking at the files changed tab, my bad.
¯\_(ツ)_/¯

octareenroon91 commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm:
```diff
@@ -214,3 +214,79 @@
+			mask = /obj/item/clothing/mask/gas/explorer
+			if(prob(20))
+				suit = pickweight(list(/obj/item/clothing/suit/hooded/explorer = 18, /obj/item/clothing/suit/hooded/cloak/goliath = 2))
+			if (belt == -1 && prob(4))
```
`belt == -1` what

Expletive replied:
See corpse.dm. The variables are initialized to -1 instead of null. Setting them allows you to override a specified outfit datum's equipment. -1 means don't change the datum's equipment for that slot, null means remove it, while a type path means it should be that path instead of what the outfit normally has.

Bawhoppen commented:
seems incomplete and not original not to be mean

FantasticFwoosh commented:
In my opinion, guranteed body drops of your special bodies (or any body) shouldn't come from common and respawnable tendrils because it just becomes a case of who can farm the legion tendril the hardest.

**All of these loot drops are too powerful**

> Golem body-drops are too strong above (night vision+ gives too powerful a loot drop), dropping random common tools instead and mesons is fine, maybe also have them drop marker beacons or liberator capes.

> Sinew belts are not properly coded and non-functional in the game, bone armor should be omitted (too strong) but rare helmets (maybe some more legionnaire helmets), and more common daggers

> I hate all the shaft miner's suggested loot drops because they all smell of power creep, none of the tools dropped should be vaguely useful, spare marker beacons, space cash etc, as well as explorer gas masks would be acceptable. (webbing & goliath cloak too useful, survival medipen is a full heal that is not needed) KA modkit is ok i guess as long as its a randomised one which makes getting a good one even rarer.
 
Stonewalling this until it get some changes.

Repukan commented:
Blacklist legions from tendrils then it'll feel balanced.

Otherwise seems ok

Repukan commented:
Also fwoosh you have no power so stop soft implying it fucking jesus

RemieRichards commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm:
```diff
@@ -214,3 +214,79 @@
+				glasses = pickweight(list(/obj/item/clothing/glasses/meson = 5, /obj/item/clothing/glasses/night = 4, /obj/item/clothing/glasses/hud/health/night = 1))
+			if(prob(20))
+				belt = pickweight(list(/obj/item/weapon/storage/belt/mining/vendor = 1,/obj/item/weapon/storage/belt/utility/full = 1))
+		if("YeOlde")
```
Battle Maid, nice.

AnturK commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm:
```diff
@@ -214,3 +214,79 @@
+				suit = /obj/item/clothing/suit/armor/bone
+				gloves = /obj/item/clothing/gloves/bracer
+			if(prob(5))
+				back = pickweight(list(/datum/crafting_recipe/bonespear = 3,/obj/item/weapon/twohanded/fireaxe/boneaxe = 2))
```
I think you wanted actual spear not the recipe here.

Expletive committed:
Rebalance, add clown corpses.

Expletive committed:
Missed a )

Expletive commented:
I've changed it a bit. The special corpses now have a combined total of 1% chance to drop. I've added a Clown category, and rebalanced the drop rates so that it is 66% miner, 11% ash walker, 11% golem, 11% clown, and 1% special. I've also modified it so that legions spawned from tendrils will drop the naked corpses as before.

ChangelingRain commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm:
```diff
@@ -214,3 +217,101 @@
+			if(prob(10))
+				r_pocket = /obj/item/weapon/implanter/sad_trombone
+		if("Golem")
+			mob_species = /datum/species/golem/random
```
sand, glass, and cloth golems shouldn't show up; do they?

coiax replied:
It depends on whether the species is applied before or after the new human is killed. But yes, none of those golems should be legion-able.

FantasticFwoosh commented:
of course i don't @Repukan but im saving time making a coding feedback thread later.

PKPenguin321 commented:
I like this for what it's worth. Maybe make it only apply to legion spawned at roundstart if tendril camping becomes an issue. The only loot that seems really bad is the claymore, since it's just a stronger esword

Repukan commented:
You can retract claymores to the hilt?

Steelpoint commented:
I should state overall I like the idea of making the legion corpses more "lifelike". 

I only want to avoid people farming Legions to get good gear.

kevinz000 commented:
Fwoosh is relevant

Expletive commented:
@PKPenguin321 I should clarify. The claymore isn't a real claymore. It's the holy  claymore. The same one the chaplain gets, not the 40 damage claymore used in highlander and such.

Expletive committed:
New Cultist corpse, adjusted drop rates, miners can have random pickaxes.

Expletive committed:
Charred skeletons from tendril spawned legion.

FantasticFwoosh commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm:
```diff
@@ -240,6 +240,8 @@
 				suit = pickweight(list(/obj/item/clothing/suit/hooded/explorer = 18, /obj/item/clothing/suit/hooded/cloak/goliath = 2))
```
The suit is better than the goliath cloak because it can be reinforced, please alter this for balance. (I don't like either cloak or suit being here but having to choose 1 of 2 to remove the suit has to go)

Expletive replied:
Yes, but the suit is also available at round start, while the cloak is not.

FantasticFwoosh replied:
Exactly but you can order more via cargo, it seems to break balance of getting extra suits.

FantasticFwoosh commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm:
```diff
@@ -266,23 +268,24 @@
-				outfit.neck = /obj/item/weapon/bedsheet/rd/royal_cape
+				neck = /obj/item/weapon/bedsheet/rd/royal_cape
 			if(prob(10))
 				l_pocket = pick(list(/obj/item/weapon/crowbar/power, /obj/item/weapon/wrench/power, /obj/item/weapon/weldingtool/experimental))
```
The advanced welding tool gives incredible tech, no thank you or weight it heavily please.

Expletive replied:
It is already weighted heavily. Golems are 1 in 10 corpses. 1 in 10 golems has a fancy tool. One in 3 of those have a welder. That comes out to one in 300 legions dropping a corpse with a welding tool. Or are you saying that 0.33% is too high of a chance?

FantasticFwoosh commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm:
```diff
@@ -218,8 +218,16 @@
+	mob_name = "ashen skeleton"
+	mob_gender = NEUTER
+	husk = FALSE
+	mob_species = /datum/species/skeleton
```
These shouldn't be revived with strange reagent, please be careful with adding skeletons without having some kind of additional code to make them basically fake.

> Transplant, insert strange reagent onto some implanted eyes.

Expletive replied:
Right, that's a good point. I originally had burn damage on them, but for some reason that gave them the husk sprite rather than the skeletons. I'll look through mob code and see if I can exlude skeletons from husking.

Expletive committed:
charred skeletons have damage and can't become husks.

ChangelingRain commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm:
```diff
@@ -240,6 +240,8 @@
 				suit = pickweight(list(/obj/item/clothing/suit/hooded/explorer = 18, /obj/item/clothing/suit/hooded/cloak/goliath = 2))
 			if (mob_species != /datum/species/plasmaman && prob(4))
 				belt = pickweight(list(/obj/item/weapon/storage/belt/mining = 2, /obj/item/weapon/storage/belt/mining/alt = 2))
+			else if(prob(10))
```
no plasmaman check?

octareenroon91 commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm:
```diff
@@ -214,3 +217,122 @@
+				belt = pickweight(list(/obj/item/weapon/storage/belt/mining = 2, /obj/item/weapon/storage/belt/mining/alt = 2))
+			else if(prob(10))
+				belt = pickweight(list(/obj/item/weapon/pickaxe = 8, /obj/item/weapon/pickaxe/mini = 4, /obj/item/weapon/pickaxe/silver = 2, /obj/item/weapon/pickaxe/diamond = 1))
+			else if (mob_species != /datum/species/plasmaman)
```
All the belt possibilities need to account for plasmaman species. Just move all the belt possibilities inside an `if (mob_species != /datum/species/plasmaman)`.

Expletive committed:
belts inside !plasmaman block
